sycl/source/detail/config.hpp

```diff
@@ -163,6 +165,32 @@ template <> class SYCLConfig<SYCL_PI_TRACE> {
   }
 };
 
+template <> class SYCLConfig<SYCL_DEVICE_TRIPLES> {
+  using BaseT = SYCLConfigBase<SYCL_DEVICE_TRIPLES>;
+
+public:
+  static device_triple_list *get() {
+    static bool Initialized = false;
+    static device_triple_list *TripleList = nullptr;
+
+    // Configuration parameters are processed only once, like reading a string
+    // from environment and converting it into a typed object.
+    if (Initialized) {
+      return TripleList;
+    }
+
+    Initialized = true;
+    const char *ValStr = BaseT::getRawValue();
+    if (ValStr) {
+      std::string TripleString = ValStr;
+      std::transform(TripleString.begin(), TripleString.end(),
+                     TripleString.begin(), ::tolower);
+      TripleList = new device_triple_list(TripleString);
```

Comment:
It looks like a memory leak. Where the memory allocate on this line is released?

Reply:
Yes, I will create a static object and return its address.